src/util-interval-tree.h

```diff
@@ -0,0 +1,54 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Shivani Bhardwaj <shivani@oisf.net>
+ */
+
+#ifndef __UTIL_INTERVAL_TREE_H__
+#define __UTIL_INTERVAL_TREE_H__
+
+#include "interval-tree.h"
+#include "detect.h"
+
+typedef struct SCIntervalNode {
```

Comment:
@inashivb the naming of the file and API suggests a generic interval tree implementation, but this type is specific to ports. I think it would be appropriate to either move this into detect-engine-port.c as a port specific thing, or generalize it to support other types of integers, so probably renaming port to low and port2 to high or something, and using uint64_t as the type?

Reply:
Indeed. Intended to rename the file for the same reasons, hadn't considered the other things. Will do.